Tests/RudderStackAnalyticsTests/StorageModuleTests.swift

```diff
@@ -5,259 +5,306 @@
 //  Created by Satheesh Kannan on 18/09/24.
 //
 
-import XCTest
+import Testing
 @testable import RudderStackAnalytics
 
 // MARK: - StorageModuleTests
-final class StorageModuleTests: XCTestCase {
-
-    var analytics_disk: Analytics?
-    var analytics_memory: Analytics?
+@Suite("Storage Module Tests")
+struct StorageModuleTests {
     
-    override func setUpWithError() throws {
-        try super.setUpWithError()
-        self.analytics_disk = MockProvider.clientWithDiskStorage
-        self.analytics_memory = MockProvider.clientWithMemoryStorage
-    }
-
-    override func tearDownWithError() throws {
-        try super.tearDownWithError()
-        self.analytics_disk = nil
-        self.analytics_memory = nil
-    }
-}
-
-// MARK: - KeyValueStore
-extension StorageModuleTests {
+    // MARK: - Test Data
+    let sampleEvent = "{\"event\":\"test_event\",\"properties\":{\"key\":\"value\"}}"
     
-    func test_write_primitive() {
-        guard let storage = self.analytics_memory?.configuration.storage else { XCTFail("Storage not initialized"); return }
-        
-        let value1 = 1
-        storage.write(value: value1, key: "IntValue")
-        let stored1: Int? = storage.read(key: "IntValue")
-        XCTAssertNotNil(stored1)
-        
-        let value2 = 2.0
-        storage.write(value: value2, key: "DoubleValue")
-        let stored2: Double? = storage.read(key: "DoubleValue")
-        XCTAssertNotNil(stored2)
-        
-        let value3 = "3"
-        storage.write(value: value3, key: "StringValue")
-        let stored3: String? = storage.read(key: "StringValue")
-        XCTAssertNotNil(stored3)
-        
-        let value4 = true
-        storage.write(value: value4, key: "BoolValue")
-        let stored4: Bool? = storage.read(key: "BoolValue")
-        XCTAssertNotNil(stored4)
+    // MARK: - Key-Value Storage Tests
+    
+    @Test("Given MockStorage when writing primitive types then values are stored and retrievable")
```

Comment:
I would recommend to follow this pattern for writing the `given.. when.. then..` statements.

```suggestion
    @Test("Given MockStorage, when writing primitive types, then values are stored and retrievable")
```

Or,

```suggestion
    @Test("given MockStorage, when writing primitive types, then values are stored and retrievable")
```